### ISSUE TITLE
FR: add input/number input view

### ISSUE THREAD

**Issue by iionly** [Elgg/Elgg#10156]
With HTML5 there's a more specific input field type for numbers avaiable. It would be nice, if Elgg would make use of it and offer an input/number view to allow for client-side validation of input value.

Maybe doing it in two steps:
1. add the input/number view to core,
2. convert core usage of input/text to input/number where necessary

Which Elgg branches would be the appropriate targets for these two steps? Step one already possible in 2.x or both steps only in master / 3.x?

**Comment by hypeJunction** [comment]
IMO unnecessary. You can pass necessary attributes to the text input view. Request variables are treated as text anyway. Number is such a loose concept - could be integer, fraction, float, telephone, credit card etc.

**Comment by iionly** [comment]
- Does text input allow for client-side validation in case an input requires a number?
- Would it work to use min, max, step and other attributes available to number inputs also for text inputs? With these additional attributes you can define/limit the expected input (e.g. only ints, number > 0 etc.).
- A number input field results in a number pad shown on mobile devices instead of the whole keyboard.
- A number input displayed a slider to increase/decrease the input.
- Number input falls back to text input if not supported by client (meaning server-side validation of input can't be omitted).

**Comment by hypeJunction** [comment]
Convincing arguments. Care to make a PR?

**Comment by iionly** [comment]
I can give it a try. Would the 2.x branch be the right one for step 1 (i.e. adding the input/number view without making any changes in already implemented usage of input/text yet)?

**Comment by hypeJunction** [comment]
Yes. New features go into 2.x.

**Comment by iionly** [comment]
PR https://github.com/Elgg/Elgg/pull/10157 adds the input/number view.

I think a separate PR for updating any existing usage of input/text to input/number in core if approriate would be best. Could such a PR also go to 2.x or would it be make to master as it would alter existing views?

**Comment by hypeJunction** [comment]
Any examples of such fields?

**Comment by iionly** [comment]
I can't say if Elgg core even has such fields that would be candidates for usage of input/number yet. I would have to look through the code first to identify such fields. If I don't find any, no further PR would be necessary. Still, the input/number would be useful for 3rd party plugins (I got the idea while working on the userpoints plugin that has quite a lot of suitable input fields).

For now I've added an example in the theme sandbox to demonstrate how it looks like.

**Comment by iionly** [comment]
I've found only one input field candidate in core that could be made a number input. This would be the input field for default_limit in views/default/forms/admin/site/update_basic.php.

Additionally, the number input type could be added to the list of input types in views/default/forms/profile/fields/add.php for adding new custom profile fields.

I also noticed that docs/guides/actions.rst contains a list of input views/types. I guess it would be necessary to add the new input/number view to the list. If yes, next question: are there any specific steps necessary when changing a doc file in a commit to update the translation templates (never done this so I thought I better ask)?
